lib/validator/task/mod.rs

```diff
@@ -1301,35 +1709,43 @@ where
     Ok(())
 }
 
+// Is this a good name? "Signal" in this context means both
+// signal receiver and signal sender
+pub struct SyncSignals<Signal: Future<Output = ()> + Send> {
```

Comment:
IMO this should be split into two structs, a `SyncSignalRx` and a `SyncSignalSenders` (trying to avoid calling it `SyncTxs` here due to `Txs` already meaning transactions and DB transactions)